newsrevenuehub/rev-engine
# DEV-1137: Update base components, add confirmation modal to payment

Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

- Updates the confirmation modal that users see when cancelling a recurring contribution to be in line with Figma linked in the story.
- Adds Modal-related base components.
- Adds size and color props to the base Button component based on Figma docs. **This affected existing UI** in that the button size used during onboarding was `extraLarge`, but we didn't know that at the time so they were left at a default size.
- Migrates existing base components to TypeScript.

#### Why are we doing this? How does it help us?

- Improved UX around cancelling recurring contributions.
- Base component changes move us closer to a consistent UI with less dev work needed (e.g. don't need to manually style modals, buttons).

#### How should this be manually tested? Please include detailed step-by-step instructions.

To test the cancellation process:

0. If need be, create a test recurring contribution.
1. Log into the contributor portal with the email address used for the recurring contribution you'll be testing.
2. Confirm that the Cancel button next to a recurring contribution looks the same as before.
3. Click the Cancel button.
4. Confirm that the modal matches the Figma spec linked in this story.
5. Click the X in the modal.
6. Confirm that the modal closes and the contribution shows as active in Stripe.
7. Click the Cancel button.
8. Click the "No, Keep Payment" button.
9. Confirm that the modal closes and the contribution shows as active in Stripe.
10. Click the Cancel button one last time.
11. Click the "Yes, Cancel" button.
12. Confirm that the modal closes, a success notification appears in the UI, and the contribution shows as cancelled in Stripe.

Note--the contribution will not change in the table after step 12. That is being handled in a separate story.

To test for visual regressions on the button changes:

1. Check that the submit button on the signup page looks correct (in particular, size).
2. Check that the submit button on the create account page looks correct.
3. Check that the submit button on the forgot password page looks correct.
4. Check that the submit button on the profile finalization modal (e.g. after you confirm your email and log into the dashboard for the first time) looks correct.

None of the functionality around any of these numbered items changed in this PR--just potentially the appearance of the button component.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

If we have screenshots of the contribution cancellation modal, they need to update.

#### Have automated unit tests been added? If not, why?

Yes.

#### How should this change be communicated to end users?

n/a

#### Are there any smells or added technical debt to note?

I created a Contribution type in the CancelRecurringButton component. I want to move this to a more general place in https://news-revenue-hub.atlassian.net/browse/DEV-489.

I had to remove prop types from our base components. It was a mistake to use MUI's prop types in the first place, unfortunately--see https://github.com/mui/material-ui/issues/14790. If there's a way to do it without having to specify all of MUI's props by hand, I'm all for it.

#### Has this been documented? If so, where?

n/a

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-1137

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

https://news-revenue-hub.atlassian.net/browse/DEV-489
